### PR TITLE
Nofruits now let you hit them with an object or alt-click to start/stop cycling. Verbs are gay but that's still an option too.

### PR DESCRIPTION
**Pull request by Intigracy**
reminder you can call verbs as procs

:cl:
- tweak: Nofruits now let you hit them with an object or alt-click to start/stop cycling. Verbs are gay but that's still an option too.

### CODE REVIEW

**Comment by Shadowmech88** [comment]
Could you also make it so that alt+clicking it with an empty hand triggers the verb?

**Comment by N3X15** [comment]
>Nofruit

**Comment by Intigracy** [comment]
yeah, why not

**Comment by Iamgoofball** [comment]
Stop

# stop

**Comment by Iamgoofball** [comment]
:cop: :stop:

**Comment by Iamgoofball** [comment]
✋✋✋✋🚫🍈

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 1]
```diff
@@ -4352,6 +4339,32 @@
 				available_snacks = shuffle(available_snacks)
 			counter++
 
+/obj/item/weapon/reagent_containers/food/snacks/pie/nofruitpie/proc/getnofruit(mob/user, obj/item/weapon/W = null)
+	if(!switching || !current_path)
+		return
+	verbs -= /obj/item/weapon/reagent_containers/food/snacks/pie/nofruitpie/verb/pick_leaf
```
Comment (Iamgoofball) [inline code/modules/reagents/reagent_containers/food/snacks.dm:4345]: Why keep the verb at all

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 2]
```diff
@@ -4352,6 +4339,32 @@
...
+	switching = 0
+	var/N = rand(1,3)
+	if(get_turf(user))
+		switch(N)
+			if(1)
+				playsound(user, 'sound/weapons/genhit1.ogg', 50, 1)
+			if(2)
+				playsound(user, 'sound/weapons/genhit2.ogg', 50, 1)
+			if(3)
+				playsound(user, 'sound/weapons/genhit3.ogg', 50, 1)
```
Comment (Iamgoofball) [inline code/modules/reagents/reagent_containers/food/snacks.dm:4355]: Can't you just 
``playsound(user, 'sound/weapons/genhit[rand (1,3)].ogg', 50, 1)``

File: code/modules/reagents/reagent_containers/food/snacks/grown.dm [hunk 3]
```diff
@@ -946,6 +932,32 @@ var/list/special_fruits = list()
 				available_fruits = shuffle(available_fruits)
 			counter++
 
+/obj/item/weapon/reagent_containers/food/snacks/grown/nofruit/proc/getnofruit(mob/user, obj/item/weapon/W)
```
Comment (Iamgoofball) [inline code/modules/reagents/reagent_containers/food/snacks/grown.dm:935]: 👮📧🍝

**Review by Iamgoofball — CHANGES_REQUESTED** [review]
Come on

File: code/modules/reagents/reagent_containers/food/snacks.dm (hunk 2 above)
Comment (Shadowmech88) [reply code/modules/reagents/reagent_containers/food/snacks.dm:4355]: I'm pretty sure BYOND doesn't allow the use of `[]` in filenames.

**Comment by Intigracy** [comment]
What needs fixing?

**Pull request merged**